docs/python_docs/python/tutorials/getting-started/crash-course/2-create-nn.md

```diff
@@ -0,0 +1,558 @@
+method`.
+
+```{.python .input  n=124}
+net_hybrid_seq.hybridize()
+```
+
+Performance
+
+To get a sense of the speedup from hybridizing, we can compare the performance
+before and after hybridizing by measuring in either case the time it takes to
+make 1000 forward passes through the network.
+
+```{.python .input  n=137}
+from time import time
+
+def bench(net, x):
+    start = time()
+    for i in range(1000):
+        y = net(x)
+    return time() - start
+
+x = np.random.normal(size=(1,512))
+
+net_hybrid_seq = nn.HybridSequential()
+
+net_hybrid_seq.add(nn.Dense(256,activation='relu'),
+        nn.Dense(128, activation='relu'),
+        nn.Dense(2)
+       )
+net_hybrid_seq.initialize()
+
+print('Before hybridizing: %.4f sec'%(bench(net_hybrid_seq, x)))
+net_hybrid_seq.hybridize()
+print('After hybridizing: %.4f sec'%(bench(net_hybrid_seq, x)))
+
+
+```
+
+```{.json .output n=137}
+[
+ {
+  "name": "stdout",
+  "output_type": "stream",
+  "text": "Before hybridizing: 0.6753 sec\nAfter hybridizing: 0.2139 sec\n"
+ }
+]
+```
+
+Peeling another layer, we also have a `HybridBlock` which is the hybrid version
+of the `Block` API.
+
+With normal Blocks, we just need to define a `forward` function that takes an
+input `x` and computes the result of the forward pass through the network. To
+define a `HybridBlock`, we create the same `forward` function. MXNet takes care
+of hybridizing the model at the backend so you don't have to make changes to
+your code to convert it to a symbolic paradigm
+
+```{.python .input  n=152}
+from mxnet.gluon import HybridBlock
+
+class MLP_Hybrid(HybridBlock):
+    def __init__(self):
+        super(MLP_Hybrid, self).__init__()
+        self.dense1 = nn.Dense(256,activation='relu')
+        self.dense2 = nn.Dense(128,activation='relu')
+        self.dense3 = nn.Dense(2)
+
+    def forward(self, x):
+        layer1 = self.dense1(x)
+        layer2 = self.dense2(layer1)
+        layer3 = self.dense3(layer2)
+        return x
+    
+net_Hybrid = MLP_Hybrid()
+net_Hybrid.initialize()
+
+print('Before hybridizing: %.4f sec'%(bench(net_Hybrid, x)))
+net_Hybrid.hybridize()
+print('After hybridizing: %.4f sec'%(bench(net_hybrid_seq, x)))
+
+
+```
+
+```{.json .output n=152}
+[
+ {
+  "name": "stdout",
+  "output_type": "stream",
+  "text": "Before hybridizing: 0.6658 sec\nAfter hybridizing: 0.2191 sec\n"
+ }
+]
+```
+
+Given a HybridBlock whose forward computation consists of going through other
+HybridBlocks, you can compile that section of the network by calling the
+HybridBlocks `.hybridize()` method.
+
+All of MXNet’s predefined layers are HybridBlocks. This means that any network
+consisting entirely of predefined MXNet layers can be compiled and run at much
+faster speeds by calling `.hybridize()`.
+
+## Next steps
+
+After you create a neural network, learn how to automatically
+compute the gradients in [Step 3: Automatic differentiation with
+autograd](3-autograd.md).
```

Comment:
How does it relate to `3-components.md`?

Reply:
hmm we are still looking at what the right flow is so please disregard that section atm. I will update it once we know which way we are going.